cmd/agent/app/builder.go

```diff
@@ -90,6 +90,7 @@ type ProcessorConfiguration struct {
 type ServerConfiguration struct {
 	QueueSize     int    `yaml:"queueSize"`
 	MaxPacketSize int    `yaml:"maxPacketSize"`
+	BufferSize    int    `yaml:"bufferSize"`
```

Comment:
I would prefer to call this SocketBufferSize, it's much more descriptive.

Similarly for the CLI flag.